### PR TITLE
[READY] Release 2.2

### PR DESCRIPTION
**Pull request by andrewgdotcom**
Update to the hockeypuck.io page to match https://github.com/hockeypuck/hockeypuck/pull/275 - since it references changes made in that PR the two should be merged (reasonably) simultaneously.

### CODE REVIEW

File: configuration.html [hunk 1]
```diff
@@ -96,103 +110,262 @@ <h2 id="TOC_1.4.">1.4. Custom HTML templates</h2>
...
+      <li><code>nworkers</code> is the number of worker threads (default 8)</li>
+    </ul>
+  </p>
+
+  <h3 id="TOC_1.3.1.">1.3.1. Storage</h3>
+  <p>
+    PostgreSQL &gt;= 9.4 is required for use with Hockeypuck, as the JSONB data type is used to store most of the public key material.
+    Some fields are broken out into separate columns for indexing.
+    For details, refer to the PostgreSQL storage backend, <a href="https://github.com/hockeypuck/hockeypuck/tree/master/src/hockeypuck/pghkp/storage.go" target="_blank">pghkp/storage.go</a>.
+    Hockeypuck has been tested with versions of PostgreSQL up to and including 15.
```
Comment (iarenaza) [inline configuration.html]: keys.escomposlinux.org. is running with PostgreSQL 16, without any apparent issues.

File: configuration.html [hunk 2]
```diff
@@ -96,103 +110,262 @@ <h2 id="TOC_1.4.">1.4. Custom HTML templates</h2>
...
-reconAddr=&#34;keys.cmarstech.com:11370&#34;
-[hockeypuck.conflux.recon.partner.peer2]
-httpAddr=&#34;juju-azure-dev-y9157oo521.cloudapp.net:11371&#34;
-reconAddr=&#34;juju-azure-dev-y9157oo521.cloudapp.net:11370&#34;</pre></div>
   <p>
-    Create a section for each peer <code>[hockeypuck.conflux.recon.partner.peername]</code>,
-    where <i>peername</i> is a unique logical name given to each peer (it doesn&#39;t have to relate
-    to the hostnames or anything).
+    <ul>
+      <li><code>host</code> is the SMTP host to connect to, in <code>host:port</code> format. (default "localhost:25")</li>
```
Comment (iarenaza) [inline configuration.html]: Is SMTPS (aka, SMTP + SSL) supported? Or SMTP + STARTTLS? I think we should mention that, saying whether they are supported or not.

File: index.html [hunk 3]
```diff
@@ -66,7 +65,7 @@ <h2 id="TOC_3.2.">3.2. SKS Reconciliation Protocol</h2>
   </p>
   <h2 id="TOC_3.3.">3.3. Modular storage backend</h2>
   <p>
-    Hockeypuck uses PostgreSQL ≥ 9.4 for storing indexed key
+    Hockeypuck uses PostgreSQL ≥ 9.4 (tested up to v15) for storing indexed key
```
Comment (iarenaza) [inline index.html]: As mentioned before, keyserver.escomposlinux.org is running with v16 without (apparent) issues.

File: populating.html [hunk 4]
```diff
@@ -34,8 +34,8 @@ <h1 id="TOC_1.">1. Populating</h1>
   </p>
   <h2 id="TOC_1.1.">1.1. Fetch a dump of recent keyfiles</h2>
   <p>
-    In Hockeypuck development and testing, I&#39;ve used the sources listed at
-    <a href="https://github.com/SKS-Keyserver/sks-keyserver/wiki/KeydumpSources" target="_blank">github.com/SKS-Keyserver/sks-keyserver/wiki/KeydumpSources</a>.
+    A partial list of current dumps is maintained
+    <a href="https://github.com/hockeypuck/hockeypuck/blob/master/contrib/data-sources.md" target="_blank">in the project contrib folder</a>.
```
Comment (iarenaza) [inline populating.html:38]: Hummm, that URL doesn't seem to exist as of today. And such a file isn't part of this pull request, is it?

File: configuration.html (hunk 1 above)
Comment (cr0bar) [reply configuration.html]: pgp.id is running on v16 as well, no issues here either.

Comment (andrewgdotcom) [reply configuration.html]: Good to know, I'll update that

File: configuration.html (hunk 2 above)
Comment (andrewgdotcom) [reply configuration.html]: This bit is commented out for now because it is untested. I'll come back to it later...

File: populating.html (hunk 4 above)
Comment (andrewgdotcom) [reply populating.html:38]: It's not, the code and the documentation are in separate repos unfortunately. Once 2.2 is merged in the main repo this file will exist.

Comment (andrewgdotcom) [reply populating.html:38]: The file is currently here: https://github.com/hockeypuck/hockeypuck/blob/20d4a5332ac2f3f8c81a6bd4ddeccf5d6531eb6b/contrib/data-sources.md

File: 1x.html [hunk 5]
```diff
@@ -27,9 +27,8 @@
   <h1 id="TOC_1.">1. Hockeypuck 1.0</h1>
   <p>
     Hockeypuck 1.0 is deprecated and available here for archival purposes only. New
-    installations of 1.0 are not recommended. Upgrading to 2.0 is strongly
-    recommended as soon as a migration path is available (this is a work in
-    progress).
+    installations of 1.0 are not recommended. Upgrading to 2.x is strongly
```
Comment (cmars) [inline 1x.html]: I would be bolder:

> Hockeypuck 1.0 has been sunset. Any deployments of 1.0 should migrate to 2.x.

File: install-source.html [hunk 6]
```diff
@@ -33,35 +33,27 @@ <h1 id="TOC_1.">1. Build and install from source</h1>
...
   <p>
-    Go will need these DVCS clients installed in order to fetch all of Hockeypuck&#39;s
-    package dependencies:
+    Go will need a Git client installed in order to fetch all of Hockeypuck&#39;s
+    package dependencies.
   </p>
-  <ul>
-    <li>Bazaar</li>
-    <li>Git</li>
-    <li>Mercurial</li>
```
Comment (cmars) [inline install-source.html:46]: This reminds me how old this project really is... back in the 1.0 times, some of the Go package dependencies were sourced from `hg` and `bzr` repos when you'd `go get`.

That was only like... 5 years ago... right? :older_man:

**Review by cmars — DISMISSED** [review]
I wonder if Markdown would be easier to write but it'd be a chore to convert and set up a build pipeline for it.

File: 1x.html (hunk 5 above)
Comment (andrewgdotcom) [reply 1x.html]: Done!

**Comment by andrewgdotcom** [comment]
> I wonder if Markdown would be easier to write but it'd be a chore to convert and set up a build pipeline for it.

At some point this will probably have to be done, but not today. :-D

**Pull request merged**